proto/proto/types/note.proto

```diff
@@ -84,11 +85,23 @@ message NoteSyncRecord {
     uint32 note_index = 1;
 
     // A unique identifier of the note which is a 32-byte commitment to the underlying note data.
-    digest.Digest note_id = 2;
+    primitives.Digest note_id = 2;
 
     // The note's metadata.
     NoteMetadata metadata = 3;
 
     // The note's inclusion proof in the block.
-    merkle.MerklePath merkle_path = 4;
+    primitives.MerklePath merkle_path = 4;
+}
+
+// List of note IDs.
+message NoteIds {
+    // List of note IDs to be queried from the database.
+    repeated primitives.Digest ids = 1;
```

Comment:
Should we have a message for a single `NoteId` (similar to how we have one for `AccountId` and `TransactionId`).